### ISSUE TITLE
Reg\Register an Operation\Page "Operation Information" - External user can register the operation

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1694]
#### Dependencies

- Blocked by #1899 
- Blocking #1895 

#### Description:
As an industry user, given that the operation has been added to my operator, I can register the operation.

[Figma for LFO or SFO operation](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=10222-8&t=HA6SzABkE1DQWPQg-4)
EIO form is handled in ticket #1962 

#### Acceptance Criteria:
Given I am on the first page of the registration form
When I click on the Operation dropdown,
Then the list of operations will only show operations that have been added but not yet registered (this dropdown was implemented in #1899 but it's not yet filtered on registration status)

Given that I selected an operation from the dropdown, 
Then the operation form of the selected operation will appear, so that I can review the information and be able to edit it

Given that I am on the operation information page of the New Registration form,
When I click on the Add a New Operation button
Then I can see an empty operation form in edit mode, so that I can fill it out

~Given that I selected EIO as my registration purpose
Then I see fewer fields on the operation information form. I only see~ 
- [ ] ~Operation name~
- [ ] ~Operation type~
- [ ] ~the multiple operators section~
~[EIO Form Figma link](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=15067-246&t=GhfSc8Nn67UW1tqz-4)~ EIO user story has been separated into ticket #1962 

#### Development Checklist:

- [x] each step of the form hits a different endpoint of the API
- [x] we can't put the add operation form as children of the multistepbase as-is because <form> within a <form> creates a hydration error. We could move the buttons outside of the form in MultiStepBase and then add form refs. Or look at how it's done for operation representative

Changes to make to Operation data model:
- [x] naics_code field: no changes required to field, but add comment that the column refers to "primary naics code"
- [x]  secondary_naics_code - add nullable field that is an FK to our naics codes table
- [x]  tertiary_naics_code - same as secondary_naics_code
- [x]  add many-to-many relationship between Operation and Registration Category (see below)

Changes to make to document_type data model:
- [ ]  add record for new_entrant_application

Changes to make to bc_obps_regulated_operation data model:

- [x]  add status field (required) which is an enum of either "Active" or "Inactive". All statuses should be "Active" by default

Create new enum class Registration Category:

- [x]  options: "Potential Reporting Operation", "Reporting Operation", "Regulated Operation", "Opt-in Operation", "New Entrant Operation", "Electricity Import Operation"
(additional data fields specific to certain registration categories will be separated out into other tickets)

Changes to make to `operation_ownership_timeline` data model (repurposing this - it turns out we don't need to track _ownership_ of an operation, but we do need to track who the "designated operator" of an operation is, which can change):
- [x] rename table to `operation_operator_timeline` (or if you can think of a better name, use that instead!)
- [x] update comments on model to reflect that the data is about the "designated"/controlling operator, who isn't necessarily the owner
- [x] fields in table are unchanged

Changes to make to `facility_ownership_timeline` data model (repurposing this - as with operations, we don't need to track facility _ownership_, but rather just what operation the facility belongs to, which can change over time):
- [x] rename table to `facility_operation_timeline` (or if you can think of a better name, use that instead!)
- [x] update comments on model to reflect that the data captures the relationship between the facility and operation it falls under, not facility ownership
- [x] fields in table are unchanged

- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by patrickisaac** [comment]
Potential to be broken down into multiple user stories

**Comment by BCerki** [comment]
We'll need to figure out how to do the snapshot thing in the db

**Comment by andrea-williams** [comment]
Update AC based on pending decisions from business area

**Comment by patrickisaac** [comment]
Splitting into multiple stories by @zoeyli-46

**Comment by shon-button** [comment]
Deletion Candidate Reason: 

Registration purpose and operation info steps have been combined into a single step

PR: https://github.com/bcgov/cas-registration/pull/1990
Ticket: https://github.com/orgs/bcgov/projects/122/views/2?filterQuery=&pane=issue&itemId=69561695

**Comment by BCerki** [comment]
> Deletion Candidate Reason:
> 
> Registration purpose and operation info steps have been combined into a single step
> 
> PR: #1990 Ticket: https://github.com/orgs/bcgov/projects/122/views/2?filterQuery=&pane=issue&itemId=69561695

I think we still need this one. I was trying not to wildly expand the scope of the one I'm working on, so I'm only adding the 'select operation' part, not the add part.

**Comment by BCerki** [comment]
@zoeyli-46 and @shon-button I've updated this ticket

**Comment by patriciarussellCAS** [comment]
My review notes: 
- I'm not sure if the list of operations I see in the dropdown are only those that have been added but not yet registered. @BCerki is there a way to confirm that this function is working correctly? I assume we want to avoid having operations that are already registered show up in this dropdown, correct? 

- Can you confirm that we have removed the following AC from this ticket? 

Given that I am on the operation information page of the New Registration form,
When I click on the Add a New Operation button
Then I can see an empty operation form in edit mode, so that I can fill it out

- I don't see a follow-up ticket this AC (for adding a new operation). Please let me know if we have this in our backlog already and I missed it. If not, I'd like to ensure we create that before I close this ticket. 
Thanks!
Tagging @zoeyli-46 for visibility

**Comment by BCerki** [comment]
> My review notes:
> 
> * I'm not sure if the list of operations I see in the dropdown are only those that have been added but not yet registered. @BCerki is there a way to confirm that this function is working correctly? I assume we want to avoid having operations that are already registered show up in this dropdown, correct?
> * Can you confirm that we have removed the following AC from this ticket?
> 
> Given that I am on the operation information page of the New Registration form, When I click on the Add a New Operation button Then I can see an empty operation form in edit mode, so that I can fill it out
> 
> * I don't see a follow-up ticket this AC (for adding a new operation). Please let me know if we have this in our backlog already and I missed it. If not, I'd like to ensure we create that before I close this ticket.
>   Thanks!
>   Tagging @zoeyli-46 for visibility

Re first *: Yes, operations that have already been registered shouldn't show in the dropdown. You can test this by registering an operation and then going back into the registration workflow. The operation you just registered shouldn't be available in the dropdown.
second *: You can add a new operation right now, but not with the button (Zoey and I have been discussing this in figma and it sounds like one of us will create a new ticket). In the meantime, to add a new operation, instead of selecting an operation in the dropdown, you can fill out the blank fields below.

**Comment by zoeyli-46** [comment]
I created this ticket [#2224](https://github.com/bcgov/cas-registration/issues/2224)

**Comment by patriciarussellCAS** [comment]
Thanks Brianna. This makes sense. I'll close this ticket now.
